packages/ui/src/components/va-tabs/VaTabs.vue

```diff
@@ -60,350 +60,401 @@
+        right,
+        center,
+        grow,
+        disabled,
+      } = props
 
-    return {
-      tabsService,
-      tabConfig,
-    }
-  })
+      return {
+        'va-tabs__container--left': left && !right && !center && !grow,
+        'va-tabs__container--right': right,
+        'va-tabs__container--center': center,
+        'va-tabs__container--grow': grow,
+        'va-tabs__container--disabled': disabled,
+      }
+    })
+
+    const computedTabsClass = computed(() => ({ 'va-tabs--vertical': props.vertical }))
 
-  created () {
-    // NOTE: this is just a temporary hack not to break everything
-    // because we need to stick to `this` for now
-    // TODO: move this logic to setup
-    this.context.tabsService = new TabsService(this)
+    const tabSelected = computed(() => valueComputed.value)
 
-    watch(() => this.$props.modelValue, () => {
-      this.updateTabsState()
+    const sliderStyles = computed(() => {
+      if (props.hideSlider) {
+        return { display: 'none' }
+      }
+
+      return {
+        backgroundColor: colorComputed.value,
+        height: props.vertical ? `${sliderHeight.value}px` : '',
+        width: props.vertical ? '' : `${sliderWidth.value}px`,
+        transform: `translateY(-${sliderOffsetY.value}px) translateX(${sliderOffsetX.value}px)`,
+        transition: animationIncluded.value ? 'var(--va-tabs-slider-wrapper-transition)' : '',
+      }
     })
-  }
 
-  get computedClass () {
-    const {
-      left,
-      right,
-      center,
-      grow,
-      disabled,
-    } = this.$props
+    const paginationControlledStyles = computed(() => {
+      // Prevents the movement of vertical tabs
+      if (props.vertical) {
+        return {
+          transform: 'translateX(0px)',
+        }
+      }
 
-    return {
-      'va-tabs__container--left': left && !right && !center && !grow,
-      'va-tabs__container--right': right,
-      'va-tabs__container--center': center,
-      'va-tabs__container--grow': grow,
-      'va-tabs__container--disabled': disabled,
-    }
-  }
+      return {
+        transform: `translateX(${startingXPoint.value - tabsContentOffset.value}px)`,
+        transition: animationIncluded.value ? 'transform ease 0.3s' : '',
+      }
+    })
 
-  get computedTabsClass () {
-    return {
-      'va-tabs--vertical': this.$props.vertical,
-    }
-  }
+    const disablePaginationLeft = computed(() => tabsContentOffset.value === 0)
 
-  get tabSelected () {
-    return this.valueComputed
-  }
+    const disablePaginationRight = computed(() => {
+      const lastTab = tabsComponents.value[tabsComponents.value.length - 1]
+      const containerClientWidth = getClientWidth(container.value)
 
-  get sliderStyles () {
-    if (this.$props.hideSlider) {
-      return { display: 'none' }
+      return (
+        lastTab?.rightSidePosition.value || 0) <= tabsContentOffset.value + containerClientWidth ||
+        (lastTab?.leftSidePosition.value || 0) <= tabsContentOffset.value
```

Comment:
If I'm not mistaken containerClientWidth can't be less than 0 so statement after || is redundant.